llvm/lib/Target/RISCV/RISCVRegisterInfo.cpp

```diff
@@ -573,6 +573,9 @@ bool RISCVRegisterInfo::eliminateFrameIndex(MachineBasicBlock::iterator II,
     int64_t Val = Offset.getFixed();
     int64_t Lo12 = SignExtend64<12>(Val);
     unsigned Opc = MI.getOpcode();
+    int64_t Imm9Val = SignExtend64<9>(Val);
```

Comment:
We can't sign extend the value to 9 bits. We need to check that the raw `Val` is uimm9.